### PR TITLE
Cart speed

### PR DESCRIPTION
**Pull request by jlamiano**

### CODE REVIEW

File: CartMoveController.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        # After relating speed to duty cycle
+        self.pwm.start(speed)
+        
+    def __del__(self):
+        self.pwm = 0
+        GPIO.cleanup()
+        
+    # Reverse Motion for t seconds
+    def reverse_motion(self, t):
```
Comment (plyons72) [inline CartMoveController.py]: Quick note, but comments below are wrong. If you're traveling in a reverse direction, the right wheel would be spinning in a counterclockwise motion, and the left wheel would be spinning in a clockwise motion when viewing the cart from above.

File: CartMoveController.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        GPIO.output(40, GPIO.LOW)
+
+        time.sleep(t)
+
+
+    #   Forward Motion for t seconds
+    #   Right spins clockwise, left counterclockwise
+    def forward_motion(self, t):
+
+        # Right Wheel Clockwise
```
Comment (plyons72) [inline CartMoveController.py:52]: These are correct... maybe comments were just copied above?

File: simple_pwm.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python3
```
Comment (plyons72) [inline simple_pwm.py:1]: Sample file to test cart speed?

**Review by plyons72 — CHANGES_REQUESTED** [review]
Some minor comment discrepancies, but great otherwise.

File: CartMoveController.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+        # Set pin high
+        GPIO.output(29, GPIO.HIGH)
+
+        # Set frequency to 1kHz
+        self.pwm = GPIO.PWM(29, 1000)
+
+        # After relating speed to duty cycle
+        self.pwm.start(speed)
+
```
Comment (G-E-N-E) [inline CartMoveController.py]: CartMoveController.py:29:0: C0303: Trailing whitespace (trailing-whitespace)

File: CartMoveController.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        # Set frequency to 1kHz
+        self.pwm = GPIO.PWM(29, 1000)
+
+        # After relating speed to duty cycle
+        self.pwm.start(speed)
+        
+    def __del__(self):
+        self.pwm = 0
+        GPIO.cleanup()
+
```
Comment (G-E-N-E) [inline CartMoveController.py]: CartMoveController.py:33:0: C0303: Trailing whitespace (trailing-whitespace)

File: CartMoveController.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import RPi.GPIO as GPIO
```
Comment (G-E-N-E) [inline CartMoveController.py]: CartMoveController.py:1:0: C0111: Missing module docstring (missing-docstring)

File: CartMoveController.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import time
+
+class CartMoveController(object):
```
Comment (G-E-N-E) [inline CartMoveController.py:4]: CartMoveController.py:4:0: C0111: Missing class docstring (missing-docstring)

File: CartMoveController.py (hunk 1 above)
Comment (G-E-N-E) [inline CartMoveController.py]: CartMoveController.py:35:4: C0103: Argument name "t" doesn't conform to snake_case naming style (invalid-name)

t is for the British!

Comment (G-E-N-E) [inline CartMoveController.py]: CartMoveController.py:35:4: R0201: Method could be a function (no-self-use)

Namespacing can be a bit weird in Python so I understand leaving it, but it's still a useful observation to be aware of.

File: CartMoveController.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
+import RPi.GPIO as GPIO
+import time
```
Comment (G-E-N-E) [inline CartMoveController.py:1]: CartMoveController.py:2:0: C0411: standard import "import time" should be placed before "import RPi.GPIO as GPIO" (wrong-import-order)

Pylint is super picky...

File: CartMoveController.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+class CartMoveController(object):
+    def __init__(self, speed):
+        self.speed = speed
```
Comment (G-E-N-E) [inline CartMoveController.py:6]: Units would be nice to help users understand what they must provide the constructor.

File: CartMoveController.py (hunk 8 above)
Comment (jlamiano) [reply CartMoveController.py:1]: Fixed in 1598fc0b70e00

**Review by plyons72 — APPROVED** [review]

**Pull request merged**